src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -250,16 +271,34 @@ ReplicateShardToAllWorkers(ShardInterval *shardInterval)
 
 		if (targetPlacement == NULL || targetPlacement->shardState != FILE_FINALIZED)
 		{
+			uint64 placementId = 0;
+
 			SendCommandListToWorkerInSingleTransaction(nodeName, nodePort, tableOwner,
 													   ddlCommandList);
 			if (targetPlacement == NULL)
 			{
-				InsertShardPlacementRow(shardId, INVALID_PLACEMENT_ID, FILE_FINALIZED, 0,
+				placementId = GetNextPlacementId();
```

Comment:
Could you add a single line comment explaining when there could be non-finalized placement for reference tables?